pkg/entrypoint/metricsbridge/metricsbridge.go

```diff
@@ -286,14 +294,31 @@ func start(cfg *cmdConfig) error {
 	logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
 	log := logrus.NewEntry(logrus.StandardLogger())
 
+	log.Info("starting metrics endpoint")
+	l, err := net.Listen("tcp", fmt.Sprintf(":%d", cfg.httpPort))
+	if err != nil {
+		return err
+	}
+
+	mux := &http.ServeMux{}
+	mux.Handle("/healthz/ready", http.HandlerFunc(readyHandler))
+	mux.Handle(cfg.metricsEndpoint, MetricsHandler())
+
+	go http.Serve(l, mux)
+
 	log.Printf("metricsbridge starting")
 
 	if cfg.configDir == "" {
 		return fmt.Errorf("config value cant be empty")
 	}
 
 	if err := run(log, cfg.configDir); err != nil {
+		metricsBridgeErrorsCounter.Inc()
```

Comment:
By putting this increment in the start function, does that mean this will only increment once? Do we not want to include an error incrementer on line 277 above as well?

Reply:
Good catch. Yes as this is right now, the counter will only ever increment once. I've moved this to a more appropriate place (inside the for loop in `run()`). Every error from `runOnce(context)` is returned to the `run()` caller. So we do not need an extra counter increment at 277.